daemon/gamemode.c

```diff
@@ -591,6 +595,7 @@ static GameModeClient *game_mode_client_new(pid_t pid, char *executable)
 		return NULL;
 	}
 	*ret = c;
+	strncpy(ret->executable, executable, PATH_MAX);
```

Comment:
`strncpy` does not ensure a terminating `\0` byte, in the theoretically case of `strlen (executable) >= PATH_MAX` the `ret->executable` would not be `\0`-terminated. So maybe `strncpy(ret->executable, executable, PATH_MAX - 1);` to be sure?

Reply:
Good point. Kind of amazed none of the warnings or even this static analysis spotted that. Might need to add more. Will follow up with fix

Reply:
364aea7 should sort that